### PR TITLE
[PKG-1250] python-cufflinks 0.17.3 :snowflake:

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Current package info**

| Channel | Downloads | Version | Platforms | License |
| --- | --- | --- | --- | --- |
| conda-forge | [Conda Downloads](https://anaconda.org/conda-forge/python-cufflinks) | [Conda Version](https://anaconda.org/conda-forge/python-cufflinks) | [Conda Platforms](https://anaconda.org/conda-forge/python-cufflinks) | Conda License |

Changelog: https://github.com/santosjorge/cufflinks/tree/master#release-notes
License: https://github.com/santosjorge/cufflinks/blob/master/license.txt
Requirements:
- https://github.com/santosjorge/cufflinks/blob/master/requirements.txt
- https://github.com/santosjorge/cufflinks/blob/master/setup.py

Actions:
1. Restructure the recipe
2. Rename the multi-output package to `python-cufflinks-suite`
3. Add two patches because `pandas.DataFrame.ix` is deprecated and does not work in newer versions.
4. Add pip check
5. Add tests.py to run tests
6. Add dev & doc urls
7. Add description

Notes:
-   The staging channel `https://staging.continuum.io/prefect/fs/colorlover-feedstock/pr2/1f8ae7c`  will be removed from `abs.yaml` before merging:

### CODE REVIEW

File: abs.yaml
Comment (M-Waszkiewicz-Anaconda) [inline abs.yaml]: staging should not be in channels ?

File: recipe/meta.yaml
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml]: I know source is saying that, but after our patch that version should be bumped up ?

**Review by M-Waszkiewicz-Anaconda — COMMENTED** [review]
possibly, pandas version should be bump up

File: abs.yaml
Comment (skupr-anaconda) [reply abs.yaml]: It has the same result

File: recipe/meta.yaml
Comment (skupr-anaconda) [reply recipe/meta.yaml]: Yeah, I'll bump it to >=1.0.0, see https://pandas.pydata.org/docs/whatsnew/v1.0.0.html

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

File: abs.yaml
Comment (boldorider4) [reply abs.yaml]: remember to remove this line before merging

**Review by cbouss — APPROVED** [review]
LGTM

File: recipe/meta.yaml
Comment (boldorider4) [inline recipe/meta.yaml:44]: I don't see a dependency on `chart-studio`

File: recipe/meta.yaml
Comment (boldorider4) [inline recipe/meta.yaml:50]: shouldn't this be `pandas>=0.19.2`?

File: recipe/83ef352c821f6f5092d4132e3ecd5d5eb6b4e9e8.patch
Comment (JeanChristopheMorinPerso) [inline recipe/83ef352c821f6f5092d4132e3ecd5d5eb6b4e9e8.patch]: Can you clarify what newer means here? In which version were these functions removed?

File: abs.yaml
Comment (skupr-anaconda) [reply abs.yaml:3]: It's here https://github.com/santosjorge/cufflinks/blob/b973f3a0f689816139de6c68855ab65ffec15567/cufflinks/plotlytools.py#L1432-L1437

File: recipe/meta.yaml
Comment (skupr-anaconda) [reply recipe/meta.yaml:44]: It's here https://github.com/santosjorge/cufflinks/blob/b973f3a0f689816139de6c68855ab65ffec15567/cufflinks/plotlytools.py#L1432-L1437

File: recipe/meta.yaml
Comment (skupr-anaconda) [reply recipe/meta.yaml:50]: Because `DataFrame.ix` was deprecated and finally removed from `pandas 1.0.0` we use `pandas` **>=1.0.0**. We apply a patch to use `DataFrame.loc` instead of `DataFrame.ix`. That's why pandas **<1.0.0** isn't suitable

File: recipe/83ef352c821f6f5092d4132e3ecd5d5eb6b4e9e8.patch
Comment (skupr-anaconda) [reply recipe/83ef352c821f6f5092d4132e3ecd5d5eb6b4e9e8.patch]: Fixed

File: abs.yaml
Comment (skupr-anaconda) [reply abs.yaml]: Done

**Review by boldorider4 — APPROVED** [review]

**Review by boldorider4 — APPROVED** [review]
LGTM when linter issues are resolved.

**Comment by skupr-anaconda** [comment]
> ERROR: clone/recipe/meta.yaml:68: missing_python_build_tool: output "cufflinks-py": Python packages require a python build tool such as setuptools.
> ERROR: clone/recipe/meta.yaml:74: missing_pip_check: For pypi packages, pip check should be present in the test commands

I would say that these errors are false positives for the multi-output packages if one of the outputs has `pin_subpackage`:
```
ERROR: clone/recipe/meta.yaml:68: missing_python_build_tool: output "cufflinks-py": Python packages require a python build tool such as setuptools.
ERROR: clone/recipe/meta.yaml:74: missing_pip_check: For pypi packages, pip check should be present in the test commands
```
Because we do not need `setuptools` to build it again and run `pip check` twice

**Pull request merged**
